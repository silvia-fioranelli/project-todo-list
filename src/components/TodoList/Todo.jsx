import { func, shape, string } from 'prop-types';
import { connect } from 'react-redux';
import {
  deleteTodo as deleteTodoAlias,
  toggleTodo as toggleTodoAlias,
} from '../../actions';

export const Todo = (props) => {
  const { item, toggleTodo, deleteTodo } = props;

  return (
    <div>
      <button
        className="btn btn-danger btn-sm mt-1 mr-1"
        onClick={() => deleteTodo(item.id)}
      >
        Delete
      </button>
      <button
        className="btn btn-success btn-sm mt-1 mr-3"
        onClick={() => toggleTodo(item.id)}
      >
        Toggle
      </button>
      <span
        style={{
          textDecoration: item.completed ? 'line-through' : '',
        }}
      >
        {item.content}
      </span>
    </div>
  );
};

Todo.propTypes = {
  deleteTodo: func.isRequired,
  toggleTodo: func.isRequired,
  item: shape({
    content: string.isRequired,
  }),
};

export const mapDispatch = {
  deleteTodo: deleteTodoAlias,
  toggleTodo: toggleTodoAlias,
};

export default connect(null, mapDispatch)(Todo);
