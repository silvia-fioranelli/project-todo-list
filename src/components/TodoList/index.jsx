import { arrayOf, func, shape } from 'prop-types';
import Todo from './Todo';

const TodoList = (props) => {
  const { items, deleteTodo, toggleTodo } = props;

  return (
    <div>
      <ol>
        {items.map((item) => (
          <li key={`todo-${item.id}`}>
            <Todo item={item} deleteTodo={deleteTodo} toggleTodo={toggleTodo} />
          </li>
        ))}
      </ol>
    </div>
  );
};

TodoList.propTypes = {
  deleteTodo: func.isRequired,
  items: arrayOf(shape({})).isRequired,
  toggleTodo: func.isRequired,
};

export default TodoList;
