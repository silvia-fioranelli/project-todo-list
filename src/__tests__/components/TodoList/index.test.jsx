import { shallow } from 'enzyme';
import TodoList from '../../../components/TodoList';

describe('TodoList', () => {
  describe('rendering', () => {
    test('should render items', () => {
      const props = {
        items: [
          { content: 'mock 1', id: 'id 1' },
          { content: 'mock 2', id: 'id 2' }
        ],
      };
      const wrapper = shallow(<TodoList {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
