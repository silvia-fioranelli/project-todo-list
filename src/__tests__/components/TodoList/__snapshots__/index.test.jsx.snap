// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList rendering should render items 1`] = `
<div>
  <Connect(Todo)
    item={
      Object {
        "content": "mock 1",
        "id": "id 1",
      }
    }
    key="todo-id 1"
  />
  <Connect(Todo)
    item={
      Object {
        "content": "mock 2",
        "id": "id 2",
      }
    }
    key="todo-id 2"
  />
</div>
`;
